easybuild/tools/build_log.py

```diff
@@ -69,9 +69,25 @@ class EasyBuildLog(fancylogger.FancyLogger):
     The EasyBuild logger, with its own error and exception functions.
     """
 
-    # self.raiseError can be set to False disable raising the exception which is
-    # necessary because logging.Logger.exception calls self.error
-    raiseError = True
+    def __init__(self, name):
+        """
+        Returns EasyBuildLog instance.
+        Definition of 'error' method is determine based on specified name; for loggers containing 'easybuild',
+        the 'error' method will raise an EasyBuildError next to logging an error
+        @param name: name for this logger instance
+        @return: EasyBuildLog instance
+        """
+        super(EasyBuildLog, self).__init__(name)
+
+        # self.raiseError can be set to False disable raising the exception which is
+        # necessary because logging.Logger.exception calls self.error
+        self.raiseError = True
+
+        # only use custom error method for EasyBuild loggers, adher to standard behaviour for other loggers
+        if 'easybuild' in name:
+            self.error = self.easybuild_error
+        else:
```

Comment:
ah, I stand corrected: `EasyBuildLog` inherits from `FancyLogger` which in turn inherits from `logging.Logger`.  you're right: the else branch is not needed and neither is the definition of `standard_error`